cpp/src/games/nim/main/UnitTests.cpp

```diff
@@ -0,0 +1,96 @@
+#include <gtest/gtest.h>
+#include <games/nim/Game.hpp>
+
+using Game = nim::Game;
+using State = Game::State;
+using StateHistory = Game::StateHistory;
+using PolicyTensor = Game::Types::PolicyTensor;
+using IO = Game::IO;
+using Rules = Game::Rules;
+using SymmetryGroup = groups::TrivialGroup;
+using GameResults = core::WinShareResults<Game::Constants::kNumPlayers>;
+
+TEST(NimGameTest, InitialState) {
+  StateHistory history;
+  history.initialize(Rules{});
+  State state = history.current();
+
+  EXPECT_EQ(Rules::get_current_player(state), 0);
+  EXPECT_EQ(state.stones_left, 21); // Assuming the game starts with 21 stones
+}
+
+TEST(NimGameTest, MakeMove) {
+  StateHistory history;
+  history.initialize(Rules{});
+  Rules::apply(history, 2);
```

Comment:
Maybe this would be slightly clearer if it was written as,

```
Rules::apply(history, kTake3);
```

You would define `const core::action_t kTake3 = 2;` in `Constants.hpp`.